src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/BackgroundService.cs

```diff
@@ -74,7 +74,10 @@ public virtual async Task StopAsync(CancellationToken cancellationToken)
             finally
             {
                 // Wait until the task completes or the stop token triggers
-                await Task.WhenAny(_executeTask, Task.Delay(Timeout.Infinite, cancellationToken)).ConfigureAwait(false);
+                var tcs = new TaskCompletionSource<object>();
+                using CancellationTokenRegistration registration = cancellationToken.Register(s => ((TaskCompletionSource<object>)s!).SetCanceled(), tcs);
+                // Do not await the _executeTask because cancelling it will throw an OperationCanceledException which we are explicitly ignoring
```

Comment:
Nit: we usually spell canceling (and canceled and canceller) with 1 L rather than 2. Cancellation is the only variant for which we use 2.